src/rust/ensogl/lib/components/src/file_browser.rs

```diff
@@ -0,0 +1,328 @@
+//! This module provides the [`FileBrowser`] component. See type documentation for more details.
+
+mod column;
+pub mod model;
+pub mod icons;
+
+use crate::prelude::*;
+
+use model::*;
+
+use crate::shadow;
+use crate::file_browser::column::Column;
+use crate::scroll_area::ScrollArea;
+use crate::selector::Bounds;
+use crate::list_view;
+
+use enso_frp as frp;
+use ensogl_core::application;
+use ensogl_core::application::Application;
+use ensogl_core::application::shortcut;
+use ensogl_core::display;
+use ensogl_core::display::shape::*;
+use std::path::PathBuf;
+use ensogl_core::display::object::ObjectOps;
+use ensogl_theme as theme;
+use ensogl_text as text;
+use ensogl_core::data::color;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+const TOOLBAR_HEIGHT      : f32 = 45.0;
+const TOOLBAR_BORDER_SIZE : f32 = 1.0;
+const CONTENT_OFFSET_Y    : f32 = TOOLBAR_HEIGHT + TOOLBAR_BORDER_SIZE;
+const PADDING             : f32 = 16.0;
+const WIDTH               : f32 = 814.0;
+const HEIGHT              : f32 = 421.0;
+const CONTENT_HEIGHT      : f32 = HEIGHT - CONTENT_OFFSET_Y;
+const SCROLL_SPACING      : f32 = 30.0;
+
+fn title_color() -> color::Rgba {
+    color::Rgba(0.439, 0.439, 0.439, 1.0)
+}
+
+
+
+// ==================
+// === Background ===
+// ==================
+
+mod background {
+    use super::*;
+
+    pub const SHADOW_PX        : f32 = 10.0;
+    pub const CORNER_RADIUS_PX : f32 = 16.0;
+
+    // This defines a rounded rectangle, filling the whole sprite, up to a padding of `SHADOW_PX` on
+    // all sides. The rectangle has a shadow and there is a gray line separating the upper part that
+    // covers the toolbar of the file browser.
```

Comment:
I would suggest to just include links to pictures on GitHub in such shape descriptions. Does that sound good to you?

Reply:
nope, because then you need to open browser to understand docs. Docs should be understandable by reading the code. Creating a very simple description / ascii art is rather simple task and we did it already in multiple places in the code.